File: server/app/schemas/project.py
from pydantic import BaseModel, ConfigDict, Field


class CreateProjectRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True, serialize_by_alias=True)

    scene_id: str = Field(alias="sceneId")
    mode: str
    title: str


class ProjectDetail(BaseModel):
    model_config = ConfigDict(populate_by_name=True, serialize_by_alias=True)

    project_id: str = Field(alias="projectId")
    scene_id: str = Field(alias="sceneId")
    scene_title: str = Field(alias="sceneTitle")
    mode: str
    title: str
    status: str
    progress: int
    current_step: str = Field(alias="currentStep")
    material_count: int = Field(alias="materialCount")
    updated_at: str = Field(alias="updatedAt")

File: core/model/src/main/kotlin/com/yingrensheng/core/model/project/Project.kt
package com.yingrensheng.core.model.project

import java.time.Instant

enum class CreationMode {
    QUICK_FILM,
    STORY_FILM,
}

enum class ProjectStatus {
    DRAFT,
    GENERATING,
    PREVIEW_READY,
    EXPORTED,
}

data class Project(
    val projectId: String,
    val title: String,
    val sceneType: String,
    val mode: CreationMode,
    val status: ProjectStatus,
    val progress: Int,
    val currentStep: String,
    val materialCount: Int,
    val moodLabel: String,
    val updatedAt: Instant,
)

